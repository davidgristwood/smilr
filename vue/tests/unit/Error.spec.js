import { createLocalVue, shallowMount } from '@vue/test-utils'
import Error from '@/components/Error.vue'

// create an extended `Vue` constructor
const localVue = createLocalVue()
import BootstrapVue from 'bootstrap-vue'
localVue.use(BootstrapVue)

describe('Error.vue', () => {
  it('renders message when passed', () => {
    let msg = 'This is an error'

    const wrapper = shallowMount(Error, {
      localVue,
      propsData: {
        message: msg
      }
    })

    expect(wrapper).toMatchSnapshot()
  })
})
