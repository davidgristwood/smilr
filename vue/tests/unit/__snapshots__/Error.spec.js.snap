// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error.vue renders message when passed 1`] = `
<b-jumbotron-stub header="Error ðŸ˜¢" headertag="h1" headerlevel="3" lead="Something bad has happened" leadtag="p" tag="div"><pre class="error-message">This is an error</pre> <br>
  <b-button-stub variant="danger" type="button" tag="button" target="_self" onclick="window.location.reload()">
    Try to Reload App
  </b-button-stub>
</b-jumbotron-stub>
`;
